src/renderer/SplatRenderer.cpp

```diff
@@ -0,0 +1,259 @@
+
+  auto& viewportData = getCurrentViewportData();
+
+  float tanFoVY = 1.f / matP[1][1];
+  float tanFoVX = tanFoVY * (1.f * viewportData.mWidth / viewportData.mHeight);
+
+  // Convert view and projection to target coordinate system. This is done in the same manner as in
+  // the original implementation:
+  // https://gitlab.inria.fr/sibr/sibr_core/-/blob/fossa_compatibility/src/projects/gaussianviewer/renderer/GaussianView.cpp#L469
+  matP  = matP * matMV;
+  matMV = glm::row(matMV, 1, -1.f * glm::row(matMV, 1));
+  matMV = glm::row(matMV, 2, -1.f * glm::row(matMV, 2));
+  matP  = glm::row(matP, 1, -1.f * glm::row(matP, 1));
+
+  // Copy frame-dependent data to GPU.
+  CUDA_SAFE_CALL(
+      cudaMemcpy(mViewCuda, glm::value_ptr(matMV), sizeof(glm::mat4), cudaMemcpyHostToDevice));
+  CUDA_SAFE_CALL(
+      cudaMemcpy(mProjCuda, glm::value_ptr(matP), sizeof(glm::mat4), cudaMemcpyHostToDevice));
+  CUDA_SAFE_CALL(
+      cudaMemcpy(mCamPosCuda, glm::value_ptr(camPos), sizeof(float) * 3, cudaMemcpyHostToDevice));
+
+  // Map the OpenGL buffer for use with CUDA or use the fallback buffer if interop failed.
+  float* image_cuda = nullptr;
+  if (!viewportData.mInteropFailed) {
+    size_t bytes;
+    CUDA_SAFE_CALL(cudaGraphicsMapResources(1, &viewportData.mImageBufferCuda));
+    CUDA_SAFE_CALL(cudaGraphicsResourceGetMappedPointer(
+        (void**)&image_cuda, &bytes, viewportData.mImageBufferCuda));
+  } else {
+    image_cuda = viewportData.mFallbackBufferCuda;
+  }
+
+  // Draw the radiance field!
+  CudaRasterizer::Rasterizer::forward(mGeomBufferFunc, mBinningBufferFunc, mImgBufferFunc, count, 3,
+      16, mBackgroundCuda, viewportData.mWidth, viewportData.mHeight, mesh.mPosCuda, mesh.mShsCuda,
+      nullptr, mesh.mOpacityCuda, mesh.mScaleCuda, scale, mesh.mRotCuda, nullptr, mViewCuda,
+      mProjCuda, mCamPosCuda, tanFoVX, tanFoVY, false, doFade, image_cuda);
+
+  // Unmap the OpenGL resource or manually copy the data.
+  if (!viewportData.mInteropFailed) {
+    CUDA_SAFE_CALL(cudaGraphicsUnmapResources(1, &viewportData.mImageBufferCuda));
+  } else {
+    CUDA_SAFE_CALL(cudaMemcpy(viewportData.mFallbackBytes.data(), viewportData.mFallbackBufferCuda,
+        viewportData.mFallbackBytes.size(), cudaMemcpyDeviceToHost));
+    glNamedBufferSubData(viewportData.mImageBuffer, 0, viewportData.mFallbackBytes.size(),
+        viewportData.mFallbackBytes.data());
+  }
+
+  // Copy image contents to frame buffer. For this, we disable the depth test, depth writing, and
+  // perform pre-multiplied alpha blending.
+  mCopyShader.Bind();
+
+  glDisable(GL_DEPTH_TEST);
+  glEnable(GL_BLEND);
+  glBlendFunc(GL_ONE, GL_ONE_MINUS_SRC_ALPHA);
+  glDepthMask(GL_FALSE);
+
+  glUniform1i(mUniforms.mWidth, viewportData.mWidth);
+  glUniform1i(mUniforms.mHeight, viewportData.mHeight);
+
+  glBindBufferBase(GL_SHADER_STORAGE_BUFFER, 0, viewportData.mImageBuffer);
+  glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
```

Comment:
In the vertex shader a full screen triangle is created.

```suggestion
  glDrawArrays(GL_TRIANGLE_STRIP, 0, 3);
```